### ISSUE TITLE
🐛 Bug Report: Usage Statistics Are Incorrect Since 1.0 Update

### ISSUE THREAD

**Issue by lvlts** [appwrite/appwrite#4560]
### 👟 Reproduction steps

When I upgraded from 0.15 to 1.0 (now 1.0.3), the usage statistics have stopped working:

- For Databases: the default usage view (30d) shows statistics for 30 days before Sept 17 (when I upgraded to 1.0) for "Objects" and "Documents". "Databases" and "Collections" show the current date correctly (with 0 stats before it).
- For Databases: the 24h usage view correctly shows for "Documents", other views are 0 valued.

Same issue happens with the 90 days usage view. 

### 👍 Expected behavior

In the databases section, when the default view (30d) for Usage is loaded, it should show 30 day history for "Databases", "Collections", "Documents" and "Objects".

### Default (30d) View

- Objects (incorrect, see date): 

<img width="998" alt="image" src="https://user-images.githubusercontent.com/32608130/197386271-ac8e8d32-27fe-450b-a28e-debfab4e74bf.png">

- Databases (correct):
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/32608130/197386307-5c6dbe0e-1ed3-4958-a17c-66a03aa9ad5b.png">

- Collections (correct, kind-of):
<img width="1067" alt="image" src="https://user-images.githubusercontent.com/32608130/197386352-4d14a86e-55d2-4713-8fe0-33d8c2cf3ae8.png">

- Documents (incorrect, see date):
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/32608130/197386371-aa69edaa-38e1-459e-b5a8-7da449a71bfc.png">

### 24h Usage View

- Documents (correct):

<img width="1014" alt="image" src="https://user-images.githubusercontent.com/32608130/197386412-191d2e2c-371c-4677-bb52-c1f70ad5614c.png">

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

Linux

### 🧱 Your Environment

I run the `docker-compose` set-up. 

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@lvlts, thanks for raising this issue! 🙏 Some questions:

1. Did you run the migrate command after upgrading? 
2. Would you be able to share the output of when you executed the migrate command?
3. What version were you on when you ran the migrate command on? 
4. If you didn't run the migrate command while on 1.0.3, would you please do so now and share the output?

**Comment by lvlts** [comment]
@stnguyen90 Thanks for the fast reply!

1. I ran the migrate command after updating from 0.15.x to 1.0.0. I did NOT run it on 1.0.3
2. I don't have the previous output.
3. 0.15.x (it was the one in the dockerhub)
4. Sure, log below:

```
Starting Data Migration to version 1.0.3
[PDO] MySQL connection restarted
Migrating Project: Appwrite (console)
Migrating Stats
Migrating Collections
Migrating Collection "_metadata"
Created new Collection "cache" collection
Created new Collection "variables" collection
Migrating Collection "audit"
Migrating Collection "abuse"
Migrating Collection "databases"
Migrating Collection "collections"
Migrating Collection "attributes"
Migrating Collection "indexes"
Migrating Collection "projects"
'_key_name' from projects: Index already exists
Migrating Collection "platforms"
Migrating Collection "domains"
Migrating Collection "keys"
'accessedAt' from keys: Attribute already exists
'sdks' from keys: Attribute already exists
'_key_accessedAt' from keys: Index already exists
Migrating Collection "webhooks"
Migrating Collection "users"
'hash' from users: Attribute already exists
'hashOptions' from users: Attribute already exists
Migrating Collection "users" Passwords
1 / 1
'_key_name' from users: Index already exists
'_key_status' from users: Index already exists
'_key_passwordUpdate' from users: Index already exists
'_key_registration' from users: Index already exists
'_key_emailVerification' from users: Index already exists
'_key_phoneVerification' from users: Index already exists
Migrating Collection "tokens"
Migrating Collection "sessions"
Migrating Collection "teams"
'_key_name' from teams: Index already exists
'_key_total' from teams: Index already exists
Migrating Collection "memberships"
'_key_userId' from memberships: Index already exists
'_key_teamId' from memberships: Index already exists
'_key_invited' from memberships: Index already exists
'_key_joined' from memberships: Index already exists
'_key_confirm' from memberships: Index already exists
Migrating Collection "functions"
Migrating Collection "functions" Variables
0 / 0
'scheduleUpdatedAt' from functions: Attribute already exists
[PDO] MySQL connection restarted
'enabled' from functions: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP COLUMN `status`; check that it exists
'_key_name' from functions: Index already exists
'_key_enabled' from functions: Index already exists
'_key_runtime' from functions: Index already exists
'_key_deployment' from functions: Index already exists
'_key_schedule' from functions: Index already exists
'_key_scheduleNext' from functions: Index already exists
'_key_schedulePrevious' from functions: Index already exists
'_key_timeout' from functions: Index already exists
Migrating Collection "deployments"
'_key_entrypoint' from deployments: Index already exists
'_key_size' from deployments: Index already exists
'_key_buildId' from deployments: Index already exists
'_key_activate' from deployments: Index already exists
Migrating Collection "builds"
Migrating Collection "executions"
'stdout' from executions: Attribute already exists
'duration' from executions: Attribute not found
'_key_trigger' from executions: Index already exists
'_key_status' from executions: Index already exists
'_key_statusCode' from executions: Index already exists
'_key_duration' from executions: Index already exists
Migrating Collection "certificates"
Migrating Collection "buckets"
'compression' from buckets: Attribute already exists
'fileSecurity' from buckets: Attribute already exists
'_key_enabled' from buckets: Index already exists
'_key_name' from buckets: Index already exists
'_key_fileSecurity' from buckets: Index already exists
'_key_maximumFileSize' from buckets: Index already exists
'_key_encryption' from buckets: Index already exists
'_key_antivirus' from buckets: Index already exists
Migrating Collection "stats"
[PDO] MySQL connection restarted
'_key_period_time' from stats: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `_key_metric`; check that it exists
[PDO] MySQL connection restarted
'_key_metric_period_time' from stats: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `_key_metric_period`; check that it exists
Migrating Collection "realtime"
Migrating Collection "cache"
Migrating Collection "files"
Migrating Collection "variables"
Migrating Databases
0 / 0
Migrating Buckets
1 / 1
Migrating Files of default (Default)
0 / 0
[PDO] MySQL connection restarted
'permissions' from buckets: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP COLUMN `permission`; check that it exists
Migrating Documents
Migrating Collection _metadata:
26 / 26
Migrating Collection audit:
5 / 5
Migrating Collection abuse:
100 / 6325
200 / 6325
300 / 6325
400 / 6325
500 / 6325
600 / 6325
700 / 6325
800 / 6325
900 / 6325
1000 / 6325
1100 / 6325
1200 / 6325
1300 / 6325
1400 / 6325
1500 / 6325
1600 / 6325
1700 / 6325
1800 / 6325
1900 / 6325
2000 / 6325
2100 / 6325
2200 / 6325
2300 / 6325
2400 / 6325
2500 / 6325
2600 / 6325
2700 / 6325
2800 / 6325
2900 / 6325
3000 / 6325
3100 / 6325
3200 / 6325
3300 / 6325
3400 / 6325
3500 / 6325
3600 / 6325
3700 / 6325
3800 / 6325
3900 / 6325
4000 / 6325
4100 / 6325
4200 / 6325
4300 / 6325
4400 / 6325
4500 / 6325
4600 / 6325
4700 / 6325
4800 / 6325
4900 / 6325
5000 / 6325
5100 / 6325
5200 / 6325
5300 / 6325
5400 / 6325
5500 / 6325
5600 / 6325
5700 / 6325
5800 / 6325
5900 / 6325
6000 / 6325
6100 / 6325
6200 / 6325
6300 / 6325
6325 / 6325
Migrating Collection databases:
0 / 0
Migrating Collection attributes:
0 / 0
Migrating Collection indexes:
0 / 0
Migrating Collection projects:
1 / 1
Migrating Collection platforms:
1 / 1
Migrating Collection domains:
2 / 2
Migrating Collection keys:
1 / 1
Migrating Collection webhooks:
0 / 0
Migrating Collection users:
1 / 1
Migrating Collection tokens:
0 / 0
Migrating Collection sessions:
5 / 5
Migrating Collection teams:
1 / 1
Migrating Collection memberships:
1 / 1
Migrating Collection functions:
0 / 0
Migrating Collection deployments:
0 / 0
Migrating Collection builds:
0 / 0
Migrating Collection executions:
0 / 0
Migrating Collection certificates:
2 / 2
Migrating Collection buckets:
1 / 1
Migrating Collection stats:
0 / 0
Migrating Collection realtime:
13 / 13
Migrating Collection cache:
11 / 11
Migrating Collection variables:
0 / 0
Clean up 'write' Permissions
Migrated 1/2 projects...
Migrating Project: MyTestProject (630a033f3e26410bbade)
Migrating Stats
Migrating Collections
Migrating Collection "_metadata"
Created new Collection "cache" collection
Created new Collection "variables" collection
Migrating Collection "audit"
Migrating Collection "abuse"
Migrating Collection "databases"
Migrating Collection "collections"
Migrating Collection "attributes"
Migrating Collection "indexes"
Migrating Collection "projects"
'_key_name' from projects: Index already exists
Migrating Collection "platforms"
Migrating Collection "domains"
Migrating Collection "keys"
'accessedAt' from keys: Attribute already exists
'sdks' from keys: Attribute already exists
'_key_accessedAt' from keys: Index already exists
Migrating Collection "webhooks"
Migrating Collection "users"
'hash' from users: Attribute already exists
'hashOptions' from users: Attribute already exists
Migrating Collection "users" Passwords
2 / 2
'_key_name' from users: Index already exists
'_key_status' from users: Index already exists
'_key_passwordUpdate' from users: Index already exists
'_key_registration' from users: Index already exists
'_key_emailVerification' from users: Index already exists
'_key_phoneVerification' from users: Index already exists
Migrating Collection "tokens"
Migrating Collection "sessions"
Migrating Collection "teams"
'_key_name' from teams: Index already exists
'_key_total' from teams: Index already exists
Migrating Collection "memberships"
'_key_userId' from memberships: Index already exists
'_key_teamId' from memberships: Index already exists
'_key_invited' from memberships: Index already exists
'_key_joined' from memberships: Index already exists
'_key_confirm' from memberships: Index already exists
Migrating Collection "functions"
Migrating Collection "functions" Variables
1 / 1
'scheduleUpdatedAt' from functions: Attribute already exists
[PDO] MySQL connection restarted
'enabled' from functions: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP COLUMN `status`; check that it exists
'_key_name' from functions: Index already exists
'_key_enabled' from functions: Index already exists
'_key_runtime' from functions: Index already exists
'_key_deployment' from functions: Index already exists
'_key_schedule' from functions: Index already exists
'_key_scheduleNext' from functions: Index already exists
'_key_schedulePrevious' from functions: Index already exists
'_key_timeout' from functions: Index already exists
Migrating Collection "deployments"
'_key_entrypoint' from deployments: Index already exists
'_key_size' from deployments: Index already exists
'_key_buildId' from deployments: Index already exists
'_key_activate' from deployments: Index already exists
Migrating Collection "builds"
Migrating Collection "executions"
'stdout' from executions: Attribute already exists
'duration' from executions: Attribute not found
'_key_trigger' from executions: Index already exists
'_key_status' from executions: Index already exists
'_key_statusCode' from executions: Index already exists
'_key_duration' from executions: Index already exists
Migrating Collection "certificates"
Migrating Collection "buckets"
'compression' from buckets: Attribute already exists
'fileSecurity' from buckets: Attribute already exists
'_key_enabled' from buckets: Index already exists
'_key_name' from buckets: Index already exists
'_key_fileSecurity' from buckets: Index already exists
'_key_maximumFileSize' from buckets: Index already exists
'_key_encryption' from buckets: Index already exists
'_key_antivirus' from buckets: Index already exists
Migrating Collection "stats"
[PDO] MySQL connection restarted
'_key_period_time' from stats: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `_key_metric`; check that it exists
[PDO] MySQL connection restarted
'_key_metric_period_time' from stats: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `_key_metric_period`; check that it exists
Migrating Collection "realtime"
Migrating Collection "cache"
Migrating Collection "files"
Migrating Collection "variables"
Migrating Databases
1 / 1
'documentSecurity' from database_1: Attribute already exists
Migrating Collections of 6314aacbaca54d48738c (test-db)
3 / 3
Migrating Documents of exposes (exposes)
100 / 107
107 / 107
Migrating Documents of price_history (price_history)
100 / 1043
200 / 1043
300 / 1043
400 / 1043
500 / 1043
600 / 1043
700 / 1043
800 / 1043
900 / 1043
1000 / 1043
1043 / 1043
Migrating Documents of image_cache (image_cache)
100 / 1907
200 / 1907
300 / 1907
400 / 1907
500 / 1907
600 / 1907
700 / 1907
800 / 1907
900 / 1907
1000 / 1907
1100 / 1907
1200 / 1907
1300 / 1907
1400 / 1907
1500 / 1907
1600 / 1907
1700 / 1907
1800 / 1907
1900 / 1907
1907 / 1907
[PDO] MySQL connection restarted
'permission' from database_1: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP COLUMN `permission`; check that it exists
Migrating Buckets
1 / 1
Migrating Files of very-images (very-images)
100 / 2293
200 / 2293
300 / 2293
400 / 2293
500 / 2293
600 / 2293
700 / 2293
800 / 2293
900 / 2293
1000 / 2293
1100 / 2293
1200 / 2293
1300 / 2293
1400 / 2293
1500 / 2293
1600 / 2293
1700 / 2293
1800 / 2293
1900 / 2293
2000 / 2293
2100 / 2293
2200 / 2293
2293 / 2293
[PDO] MySQL connection restarted
'permissions' from buckets: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP COLUMN `permission`; check that it exists
Migrating Documents
Migrating Collection _metadata:
30 / 30
Migrating Collection audit:
100 / 5020
200 / 5020
300 / 5020
400 / 5020
500 / 5020
600 / 5020
700 / 5020
800 / 5020
900 / 5020
1000 / 5020
1100 / 5020
1200 / 5020
1300 / 5020
1400 / 5020
1500 / 5020
1600 / 5020
1700 / 5020
1800 / 5020
1900 / 5020
2000 / 5020
2100 / 5020
2200 / 5020
2300 / 5020
2400 / 5020
2500 / 5020
2600 / 5020
2700 / 5020
2800 / 5020
2900 / 5020
3000 / 5020
3100 / 5020
3200 / 5020
3300 / 5020
3400 / 5020
3500 / 5020
3600 / 5020
3700 / 5020
3800 / 5020
3900 / 5020
4000 / 5020
4100 / 5020
4200 / 5020
4300 / 5020
4400 / 5020
4500 / 5020
4600 / 5020
4700 / 5020
4800 / 5020
4900 / 5020
5000 / 5020
5020 / 5020
Migrating Collection abuse:
3 / 3
Migrating Collection databases:
1 / 1
Migrating Collection attributes:
15 / 15
Migrating Collection indexes:
7 / 7
Migrating Collection projects:
0 / 0
Migrating Collection platforms:
0 / 0
Migrating Collection domains:
0 / 0
Migrating Collection keys:
0 / 0
Migrating Collection webhooks:
0 / 0
Migrating Collection users:
2 / 2
Migrating Collection tokens:
0 / 0
Migrating Collection sessions:
46 / 46
Migrating Collection teams:
0 / 0
Migrating Collection memberships:
0 / 0
Migrating Collection functions:
1 / 1
Migrating Collection deployments:
1 / 1
Migrating Collection builds:
1 / 1
Migrating Collection executions:
0 / 0
Migrating Collection certificates:
0 / 0
Migrating Collection buckets:
1 / 1
Migrating Collection stats:
100 / 3541
200 / 3541
300 / 3541
400 / 3541
500 / 3541
600 / 3541
700 / 3541
800 / 3541
900 / 3541
1000 / 3541
1100 / 3541
1200 / 3541
1300 / 3541
1400 / 3541
1500 / 3541
1600 / 3541
1700 / 3541
1800 / 3541
1900 / 3541
2000 / 3541
2100 / 3541
2200 / 3541
2300 / 3541
2400 / 3541
2500 / 3541
2600 / 3541
2700 / 3541
2800 / 3541
2900 / 3541
3000 / 3541
3100 / 3541
3200 / 3541
3300 / 3541
3400 / 3541
3500 / 3541
3541 / 3541
Migrating Collection realtime:
0 / 0
Migrating Collection cache:
45 / 45
Migrating Collection variables:
0 / 0
Clean up 'write' Permissions
Migrated 2/2 projects...
Data Migration Completed

```

**Comment by stnguyen90** [comment]
@lvlts, are the stats looking any better?.

If not, can you share the docker logs from the worker usage containers? There should be a database and a timeseries one. I can't recall their actual names.

**Comment by lvlts** [comment]
hi @stnguyen90 

I think you solved it :-)

I was using a `docker-compose.yml` file which had only the `usage` container. I upgraded appwrite by incrementing version numbers in my existing `docker-compose.yml` file.

When I checked `https://appwrite.io/install/compose`, I noticed the two workers there, and copied them to my existing compose configuration. My usage stats now show today as well.

Thank you for your help, closing!
